sa/database.go

```diff
@@ -223,11 +223,12 @@ func adjustMySQLConfig(conf *mysql.Config) error {
 	// `?max_statement_time=2`. A zero value in the DSN means these won't be
 	// sent on new connections.
 	if conf.ReadTimeout != 0 {
-		// In MariaDB, max_statement_time and long_query_time are both seconds.
+		// In MariaDB, max_statement_time and long_query_time are both seconds,
+		// but can have up to microsecond granularity.
 		// Note: in MySQL (which we don't use), max_statement_time is millis.
 		readTimeout := conf.ReadTimeout.Seconds()
-		setDefault("max_statement_time", fmt.Sprintf("%g", readTimeout*0.95))
-		setDefault("long_query_time", fmt.Sprintf("%g", readTimeout*0.80))
+		setDefault("max_statement_time", fmt.Sprintf("%f", readTimeout*0.95))
+		setDefault("long_query_time", fmt.Sprintf("%f", readTimeout*0.80))
```

Comment:
```suggestion
		setDefault("max_statement_time", fmt.Sprintf("%.6f", readTimeout*0.95))
		setDefault("long_query_time", fmt.Sprintf("%.6f", readTimeout*0.80))
```
I'd explicitly specify the precision we want, rather than letting the go stdlib use the default that happens to match what we want, so that future readers have a chance at understanding that this format verb is load-bearing.

Reply:
I definitely considered that, but thought, "Why would I override this to what the default is set to?" 🤷🏼‍♂️